src/posts/create.js

```
@@ -32,6 +35,21 @@ module.exports = function (Posts) {
 		const pid = data.pid || await db.incrObjectField('global', 'nextPid');
 		let postData = { pid, uid, tid, content, sourceContent, timestamp, isEnglish, translatedContent };
 
+		// Start translation in background (non-blocking)
+		translate.translate(data).then(([detected, translated]) => {
+			// Update post asynchronously when translation completes
+			Posts.setPostFields(pid, {
+				isEnglish: detected,
+				translatedContent: translated,
```

Nice: Silent fail with error logging is appropriate here since the post is already created successfully. Consider emitting a socket event when translation completes so the UI can update without page refresh.